Landmark typography updates (#5575)
grommet/hpe-design-system: aries-site/src/pages/foundation/typography.mdx

    }}
    width="100%"
  >
    <SyntaxHighlighterWrapper tabIndex="0" wrapLongLines language="javascript">
      {`<h1 class="heading-xlarge">Page title</h1>
<h2 class="card heading-medium">Card title</h2>
<h2 class="heading-large">Section title</h2>
<h3 class="card heading-medium">Another card title</h3>`}
    </SyntaxHighlighterWrapper>
  </Example>
  <Example
    height="none"
    bestPractice={{
      type: 'dont',
      message: `Change the semantic level of a heading to adjust the visual styling. This can lead to accessibility issues and break the semantic structure of the page. Notice the incorrect jump from h1 to h3.`,
    }}
  >
    <SyntaxHighlighterWrapper tabIndex="0" wrapLongLines language="javascript">
      {`<h1 class="heading-xlarge">Page title</h1>
<h3 class="card heading-medium">Card title</h2>
<h2 class="heading-large">Section title</h2>
<h3 class="card heading-medium">Another card title</h3>`}
    </SyntaxHighlighterWrapper>
  </Example>
</BestPracticeGroup>

### Styling exceptions

> Some components may require visual consistency regardless of heading level. For example, card titles in dashboard layouts often need to share the same text size even if their semantic levels differ. In these cases, keep the correct semantic heading level, but apply the appropriate [token to style the text](/design-tokens/typography-system#text).

Reviewer: ```suggestion
Some components may require visual consistency regardless of heading level. For example, card titles in dashboard layouts often need to share the same text size even if their semantic levels differ. In these cases, keep the correct semantic heading level, but apply the appropriate [design token to style the text](/design-tokens/typography-system#text).
```

- For Card specific guidance, see [card titles](/components/card#card-titles) documentation.
- For Layer specific guidance, see [layer titles](/components/layer#layer-titles) documentation.

> ### Heading levels in Grommet

Reviewer: I think we should avoid calling out / writing Grommet specific guidance.

We need to move towards a direction where we provide the ability for readers to select their tech of choice and the example is presented in that syntax.

Author: Changed title and added comment for future enhancement

<h2 class="heading-large">Section title</h2>
<h3 class="card heading-medium">Another card title</h3>`}
    </SyntaxHighlighterWrapper>
  </Example>
  <Example
    height="none"
    bestPractice={{
      type: 'dont',
      message: `Change the semantic level of a heading to adjust the visual styling. This can lead to accessibility issues and break the semantic structure of the page. Notice the incorrect jump from h1 to h3.`,
    }}
  >
    <SyntaxHighlighterWrapper tabIndex="0" wrapLongLines language="javascript">
      {`<h1 class="heading-xlarge">Page title</h1>
<h3 class="card heading-medium">Card title</h2>
<h2 class="heading-large">Section title</h2>
<h3 class="card heading-medium">Another card title</h3>`}
    </SyntaxHighlighterWrapper>
  </Example>
</BestPracticeGroup>

### Styling exceptions

Some components may require visual consistency regardless of heading level. For example, card titles in dashboard layouts often need to share the same text size even if their semantic levels differ. In these cases, keep the correct semantic heading level, but apply the appropriate [token to style the text](/design-tokens/typography-system#text).

> 

Reviewer 2: I would move this content here:
```suggestion
- For Card specific guidance, see [card titles](/components/card#card-titles) documentation.
- For Layer specific guidance, see [layer titles](/components/layer#layer-titles) documentation.
```

> ### Heading levels in practice

Reviewer 2: ```suggestion
### Heading levels in Grommet
```

I also wonder if this needs to be a section or if it fits somewhere else?

Reviewer 2: I had understood that the implementation that was described here was specific for grommet consumers. If that's not the case I am happy with your suggestion.

Use `level={1}` for the single `<h1>` on the page. Follow with `level={2}` for main sections and `level={3}` for subsections. See the example below for a recommended structure.

> It is uncommon to use a heading level higher than 3, as this often indicates that the page content is very dense. Before implementing heading levels 4–6, consider whether the information can be presented more concisely or divided into additional sections.

Reviewer 2: ```suggestion
```

moved this content above since it's more "best practice" than "practice"

```

## Paragraph

> Paragraphs represent blocks of flowing text and are styled using [Text design tokens](/design-tokens/typography-system#text) for font size, weight, and line height. To ensure optimal readability, Paragraphs differ from standard text by enforcing a maximum width to limit line length. Implementations should reference the corresponding Text t-shirt size tokens for typography properties and apply the associated `maxWidth` value to the container.

Reviewer: "apply the associated `maxWidth` value to the container" <-- this technique is too particular to a specific tool.

```suggestion
Paragraphs represent blocks of flowing text and are styled using [Text design tokens](/design-tokens/typography-system#text) for font size, weight, and line height. Paragraphs differ from standard text by providing a maximum width [limiting line length for optimal readability](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-visual-presentation.html). Implementations should reference the corresponding Text t-shirt sizes and maxWidth design tokens.
```

  <TextSizeExample />
</Example>

### Text best practices

Use the t-shirt size `maxWidth` design token on paragraphs to limit the width of text to ensure readability.

While each text t-shirt sizes comes with a default font weight, the emphasis of text can be adjusted to suit the context by using the [fontWeight](/design-tokens/global#fontweight) design tokens.

## Responsiveness

The values of heading tokens automatically step down one t-shirt size at the `small` [breakpoint](/design-tokens/global#breakpoint). This ensures content scales appropriately as the screen size decreases.

> When utilizing the CSS files, a media query is included in the "dimension.small.css" file to adjust the token values at the `small` breakpoint.

Reviewer: This is too specific to a particular implementation.

I would move this a content under `design-tokens/global#breakpoint` either as inline content or as a "Tip" inline notification (but @julauxen is trying to avoid the latter).
